conf/nginx/nginx.conf

```
@@ -0,0 +1,132 @@
+events {
+  worker_connections  4096;
+}
+
+http {
+
+    include /etc/nginx/mime.types;
+    default_type    application/octet-stream;
+    client_max_body_size 500M;
+    ssl_ciphers         EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH;
+    ssl_dhparam         /etc/ssl/dhparam.pem;
+    ssl_prefer_server_ciphers on;
+    ssl_protocols       TLSv1.2;
+    ssl_session_cache   shared:SSL:10m;
+    ssl_session_timeout 10m;
+
+    gzip on;
+    gzip_comp_level 6;
+    gzip_vary on;
+    gzip_types text/plain text/css application/json application/x-javascript application/javascript text/xml application/xml application/rss+xml text/javascript image/svg+xml application/vnd.ms-fontobject application/x-font-ttf font/opentype;
+
+    charset  utf-8;
+
+    # Extra slashes matter to Agave
+    merge_slashes off;
+
+    upstream portal_cms {
+        server core_portal_cms:8000;
+    }
+
+    upstream portal_core {
+        server core_portal_django:6000;
+    }
+
+    upstream portal_ws {
+        server core_portal_websockets:9000;
+    }
+
+    ## Include any custom http directives
+    include /etc/nginx/conf.d/*.http.conf;
+
+    server {
+        listen 80 default_server;
+        listen [::]:80 default_server;
+        server_name _;
+        return 301 https://$host$request_uri;
+    }
+
+    server {
+        listen 443 http2 ssl;
+        ssl_certificate     /etc/ssl/certs/portal.cer;
+        ssl_certificate_key /etc/ssl/private/portal.key;
+
+        location /media  {
+            alias /var/www/portal/cms/media;
+        }
+
+        location /static {
+            alias /var/www/portal/cms/static;
+        }
+
+        location / {
+            uwsgi_read_timeout 60s;
+            uwsgi_send_timeout 60s;
+            uwsgi_pass  portal_cms;
+            include     /etc/nginx/uwsgi_params;
+        }
+
+        location /core/media  {
+            alias /var/www/portal/portal/media;
+        }
+
+        location /core/static {
+            alias /var/www/portal/portal/static;
+        }
+
+        location /core {
+            uwsgi_read_timeout 60s;
+            uwsgi_send_timeout 60s;
+            uwsgi_pass  portal_core;
+            include     /etc/nginx/uwsgi_params;
+        }
+
+        location ~ ^/(auth|workbench|tickets|accounts|api|login|webhooks|googledrive-privacy-policy|public-data|search) {
```

```suggestion
        location ~ ^/(core/markup|auth|workbench|tickets|accounts|api|login|webhooks|googledrive-privacy-policy|public-data|search) {
```
We need to tell nginx to look for `core/markup/...` in the Django container, otherwise it plasters the nginx 502 page on the screen in CMS pages.

---

would the block starting at line 77 (`location /core`) cover that?